src/screens/Ticketing/Purchase/utils.ts

```diff
@@ -16,19 +32,35 @@ export type PurchaseFlow = {
   travelDateSelectionEnabled: boolean;
 };
 
+export type UserProfileTypeWithCount = {
+  userTypeString: string;
+  count: number;
+};
+
 export const getPurchaseFlow = (
   product: PreassignedFareProduct,
 ): PurchaseFlow => {
-  if (product.type === 'period' || product.type === 'hour24') {
-    return {
-      travellerSelectionMode: 'single',
-      travelDateSelectionEnabled: true,
-    };
-  } else {
-    return {
-      travellerSelectionMode: 'multiple',
-      travelDateSelectionEnabled: false,
-    };
+  switch (product.type) {
+    case 'period':
+      return {
+        travellerSelectionMode: 'single',
+        travelDateSelectionEnabled: true,
+      };
+    case 'hour24':
+      return {
+        travellerSelectionMode: 'single',
+        travelDateSelectionEnabled: true,
+      };
+    case 'single':
+      return {
+        travellerSelectionMode: 'multiple',
+        travelDateSelectionEnabled: false,
+      };
+    case 'carnet':
+      return {
+        travellerSelectionMode: 'single',
+        travelDateSelectionEnabled: false,
+      };
```

Comment:
Carnet is not available for purchase in the app, but added for completeness sake, instead of having a fallback. Also means the same function could be used in the webshop.